client/app/dedicatedCloud/security/kms/edit/dedicatedCloud-security-kms-edit.controller.js

```diff
@@ -0,0 +1,87 @@
+angular.module('App').controller('DedicatedCloudSecurityKMSEditCtrl', class DedicatedCloudSecurityKMSEditCtrl {
+  constructor(
+    $stateParams, $timeout, $translate, $uibModalInstance,
+    DedicatedCloud, VM_ENCRYPTION_KMS, kmsToEdit,
+  ) {
+    this.$timeout = $timeout;
+    this.$translate = $translate;
+    this.$uibModalInstance = $uibModalInstance;
+    this.DedicatedCloud = DedicatedCloud;
+    this.VM_ENCRYPTION_KMS = VM_ENCRYPTION_KMS;
+
+    this.serviceName = $stateParams.productId;
+    this.kmsToEdit = kmsToEdit;
+  }
+
+  $onInit() {
+    this.editTaskId = null;
+    this.kmsEditionTask = {
+      name: null,
+      state: null,
+      progress: 0,
+      description: null,
+    };
+
+    this.pollRequestPending = false;
+  }
+
+  closeModal() {
+    this.$uibModalInstance.dismiss();
+  }
+
+  editKms() {
+    this.kmsEditionTask.name = this.$translate.instant('dedicatedCloud_vm_encryption_edit_kms_editing');
+    this.kmsEditionTask.state = this.$translate.instant('dedicatedCloud_vm_encryption_kms_pending_task_state');
+    return this.DedicatedCloud.editVMEncryptionKMS(this.serviceName, this.kmsToEdit)
+      .then(({ taskId }) => {
+        this.editTaskId = taskId;
+        this.pollEditionTask();
+      })
+      .catch((error) => {
+        this.error = error;
+      });
+  }
+
+  pollEditionTask() {
+    if (this.editTaskId && !this.pollRequestPending && !this.isTaskFinishedOrCanceled()) {
+      this.pollRequestPending = true;
+      this.DedicatedCloud.startVMEncryptionKMSPoller(this.serviceName, this.editTaskId)
+        .then((taskDetails) => {
+          this.kmsEditionTask = taskDetails;
+        }).catch(() => {
+          this.stopEditionPoller();
+        }).finally(() => {
+          this.$timeout(() => {
+            this.pollRequestPending = false;
+          }, this.VM_ENCRYPTION_KMS.pollingDelay);
+        });
+    }
+  }
+
+  isFormValid() {
+    return _(this.kmsEditForm).get('$valid');
```

Comment:
It doesn't look like `typeof this.kmsEditForm` can be `!== "object"`

Reply:
Hmm what about this so ? https://github.com/ovh-ux/ovh-manager-dedicated/pull/682#discussion_r222635873

Reply:
The link doesn't target anything relevant 😅

Reply:
In fact, this.kmsEditForm can be undefined (at initialization), so `get` handle this case